src/environmentd/tests/server.rs

```diff
@@ -4667,3 +4669,56 @@ fn test_builtin_connection_alterations_are_preserved_across_restarts() {
         );
     }
 }
+
+#[mz_ore::test]
+#[cfg_attr(miri, ignore)] // too slow
+fn test_webhook_request_compression() {
+    let server = test_util::TestHarness::default().start_blocking();
+    let mut client = server.connect(postgres::NoTls).unwrap();
+
+    // Create a webhook source.
+    client
+        .execute(
+            "CREATE CLUSTER webhook_cluster REPLICAS (r1 (SIZE '1'));",
+            &[],
+        )
+        .expect("failed to create cluster");
+    client
+        .execute(
+            "CREATE SOURCE webhook_text IN CLUSTER webhook_cluster FROM WEBHOOK BODY FORMAT TEXT",
+            &[],
+        )
+        .expect("failed to create source");
+
+    let http_client = Client::new();
+    let webhook_url = format!(
+        "http://{}/api/webhook/materialize/public/webhook_text",
+        server.inner().http_local_addr(),
+    );
+
+    let og_body = "hello world!";
+
+    let mut encoder = GzEncoder::new(Vec::new(), Compression::default());
+    encoder.write_all(og_body.as_bytes()).unwrap();
+    let compressed_body = encoder.finish().unwrap();
+
+    assert!(og_body.as_bytes().len() < compressed_body.len());
+
+    let resp = http_client
+        .post(&webhook_url)
+        .header(CONTENT_ENCODING, "gzip")
+        .body(compressed_body)
+        .send()
+        .expect("failed to POST event");
+    assert!(resp.status().is_success(), "{resp:?}");
+
+    // Wait a moment for the webhook to get appended.
+    std::thread::sleep(std::time::Duration::from_secs(5));
```

Comment:
Would be nicer to retry until we get the correct result (or even better, have this test in testdrive)

Reply:
I've added a loop that retries up to 10 times. I don't think putting this in testdrive is possible, since the test needs to be quite specific about how the HTTP request is formed.